packages/@azure/servicebus/data-plane/lib/core/messageReceiver.ts

```diff
@@ -761,24 +761,40 @@ export class MessageReceiver extends LinkEntity {
           _receiver: this._receiver
         };
         log.error(
-          "[%s] Something is busted. State of Receiver '%s' with address '%s' is: %O",
+          "[%s] Something went wrong. State of Receiver '%s' with address '%s' is: %O",
           connectionId,
           this.name,
           this.address,
           state
         );
       }
       if (shouldReopen) {
+        log.receiver(
+          "[%s] Closing the disconnected [%s]Receiver and creating a new one..",
+          this._context.namespace.connectionId,
+          this.name
+        );
         // provide a new name to the link while re-connecting it. This ensures that
         // the service does not send an error stating that the link is still open.
         const options: ReceiverOptions = this._createReceiverOptions(true);
         // shall retry forever at an interval of 15 seconds if the error is a retryable error
         // else bail out when the error is not retryable or the oepration succeeds.
+        log.receiver(
+          "[%s] New [%s]Receiver has been created.",
+          this._context.namespace.connectionId,
+          this.name
+        );
         const config: RetryConfig<void> = {
-          operation: () => this._init(options),
+          operation: () =>
+            this._init(options).then(() => {
+              if (this._receiver) {
+                this._receiver.addCredit(this.maxConcurrentCalls);
+              }
```

Comment:
This is very specific to the streaming receiver. In the `if` condition, can you check for the type? In the future, if we ever want to recreate the batching receiver, then we shouldnt be adding credits this way.